tandemhunter-main/TandemHunter.py

```diff
@@ -0,0 +1,465 @@
+            # to call _process_file) for compatibility with python2.7
+            results = pool.map(self, [file for file in files])
+            pool.close()
+
+        # print error if neither file or batch given
+        else:
+            sys.stderr.write(
+                "[{0}] Error: File or batch folder not specified!".format(
+                    datetime.now()
+                )
+            )
+            sys.exit()
+
+    @classmethod
+    def _find_coverage_files(cls, batch_dir, cov_file_pattern):
+        """ Method to find coverage files in a directory.
+        :param: batch_dir: Path to batch directory
+        :param: cov_file_pattern: Filename suffix to identify coverage files
+        :returns: A list of coverage files found in given directory
+        """
+        print(
+            "[{0}] Finding coverage files in {1}".format(
+                datetime.now(), batch_dir
+            )
+        )
+
+        try:
+            coverage_files = glob.glob(
+                os.path.join(batch_dir, "**", "*{0}".format(cov_file_pattern)),
+                recursive=True,
+            )
+        # handle when glob recursive not available in older python versions
+        except TypeError:
+            coverage_files = []
+            for root, dirnames, filenames in os.walk(batch_dir):
+                for filename in fnmatch.filter(
+                        filenames, "*{0}".format(cov_file_pattern)):
+                    coverage_files.append(os.path.join(root, filename))
+        except Exception as e:
+            coverage_files = []
+        return coverage_files
+
+    @classmethod
+    def _import_coverage(cls, file, column_dtypes, sep="\t"):
+        """ Method to read output file from coverage file into a dataframe.
+        :param: file: Path to coverage file
+        :param: column_dtypes: Dictionary with column name and datatype for
+                            the given coverage file type
+        :param: sep: seperate used in coverage file (default is tab)
+        :returns: reader iterable containing Pandas dataframe.
+        """
+        reader = pd.read_csv(
+            file,
+            sep=sep,
+            header=0,
+            skipinitialspace=True,
+            na_values="-",
+            dtype=column_dtypes,
+            chunksize=10000,
+        )
+
+        return reader
+
+    @classmethod
+    def _import_intervals(cls, intervals, dup_threshold):
+        """ Method to process interval json file and store as a dictionary
+        :param: file: Path to interval json
+        :param: dup_threshold: Threshold used to identify dup/amp regions
+        :returns: Dictionary
+        """
+        # open json
+        with open(intervals) as json_file:
+            intervals = json.load(json_file)
+
+        # validation json contents
+        for interval in intervals:
+            [interval[k] for k in ["region1", "region2"]]
+            # add on default dup threshold if it has not been defined

```

Comment:
Lower-priority: This list comprehension might not be the best solution, as it wastefully creates an unused list - it doesn't get assigned. I'm not digging into it right now in the interests of time but better ways to achieve the same effect are discussed here: https://stackoverflow.com/questions/14249355/python-list-comprehension-without-assignment